seed/views/portfoliomanager.py

```diff
@@ -82,36 +87,55 @@ def report(self, request):
+                            'message': 'Invalid template formulation during portfolio manager data import'
+                        },
+                        status=status.HTTP_400_BAD_REQUEST
+                    )
+                if template['z_seed_child_row']:
+                    content = pm.generate_and_download_child_data_request_report(template)
+                else:
+                    content = pm.generate_and_download_template_report(template)
+            except PMExcept as pme:
                 return JsonResponse(
-                    {'status': 'error', 'message': 'Invalid template formulation during portfolio manager data import'},
+                    {'status': 'error', 'message': pme.message},
                     status=status.HTTP_400_BAD_REQUEST
                 )
-            if template['z_seed_child_row']:
-                content = pm.generate_and_download_child_data_request_report(template)
-            else:
-                content = pm.generate_and_download_template_report(template)
-        except PMExcept as pme:
-            return JsonResponse(
-                {'status': 'error', 'message': pme.message},
-                status=status.HTTP_400_BAD_REQUEST
-            )
-        try:
-            content_object = xmltodict.parse(content)
-        except Exception:  # catch all because xmltodict doesn't specify a class of Exceptions (just ParsingInterrupted)
-            return JsonResponse({'status': 'error', 'message': 'Malformed XML from template download'}, status=500)
-        try:
-            properties = content_object['report']['informationAndMetrics']['row']
-        except KeyError:
-            return JsonResponse(
-                {
-                    'status': 'error', 'message':
-                    'Processed template successfully, but missing keys -- is the template empty on Portfolio Manager?'
-                },
-                status=status.HTTP_400_BAD_REQUEST
-            )
+            try:
+                content_object = xmltodict.parse(content)
+            except Exception:  # catch all because xmltodict doesn't specify a class of Exceptions
+                return JsonResponse(
+                    {'status': 'error', 'message': 'Malformed XML from template download'},
+                    status=status.HTTP_400_BAD_REQUEST)
+            try:
+                possible_properties = content_object['report']['informationAndMetrics']['row']
+                if isinstance(possible_properties, list):
+                    properties = possible_properties
+                else:  # OrderedDict, anything else
+                    return JsonResponse(
+                        {
+                            'status': 'error',
+                            'message': 'Property list was not a list...was a preview report template used on accident?'
+                        },
+                        status=status.HTTP_400_BAD_REQUEST
+                    )
```

Comment:
This little error block is an important addition.  A key issue being hit was that SF was (and even I was) accidentally selecting the wrong template in the list because of a sort-of duplicated name.  The one being selected was set on ESPM as a "Preview", which meant the data didn't come back as a list of properties at this report->informationAndMetrics->row key level.  Instead, it came back as a dict for a single property.  I'm using this type inference as a check for whether this request is based on a preview, and warning the user if so.